GliaWidgets/Sources/Coordinators/Chat/ChatCoordinator.swift

```diff
@@ -51,15 +53,84 @@ class ChatCoordinator: SubFlowCoordinator, FlowCoordinator {
+        self.shouldStartWithTranscript = shouldStartWithTranscript
     }
 
     func start() -> ChatViewController {
         let viewController = makeChatViewController()
         return viewController
     }
 
-    // swiftlint:disable function_body_length
     private func makeChatViewController() -> ChatViewController {
+        let model: SecureConversations.ChatWithTranscriptModel =
+            shouldStartWithTranscript ? .transcript(transcriptModel()) : .chat(chatModel())
+        return ChatViewController(viewModel: model, viewFactory: viewFactory)
+    }
+
+    private func presentMediaPickerController(
+        with pickerEvent: ObservableValue<MediaPickerEvent>,
+        mediaSource: MediaPickerViewModel.MediaSource,
+        mediaTypes: [MediaPickerViewModel.MediaType]
+    ) {
+        let viewModel = MediaPickerViewModel(
+            pickerEvent: pickerEvent,
+            mediaSource: mediaSource,
+            mediaTypes: mediaTypes
+        )
+        viewModel.delegate = { [weak self] event in
+            switch event {
+            case .finished:
+                self?.mediaPickerController = nil
+            }
+        }
+
+        let controller = MediaPickerController(viewModel: viewModel)
+        mediaPickerController = controller
+        controller.viewController { [weak self] viewController in
+            self?.navigationPresenter.present(viewController)
+        }
+    }
+
+    private func presentFilePickerController(with pickerEvent: ObservableValue<FilePickerEvent>) {
+        let viewModel = FilePickerViewModel(pickerEvent: pickerEvent)
+        viewModel.delegate = { [weak self] event in
+            switch event {
+            case .finished:
+                self?.filePickerController = nil
+            }
+        }
+
+        let controller = FilePickerController(viewModel: viewModel)
+        filePickerController = controller
+        navigationPresenter.present(controller.viewController)
+    }
+
+    private func presentQuickLookController(with file: LocalFile) {
+        let viewModel = QuickLookViewModel(file: file)
+        viewModel.delegate = { [weak self] event in
+            switch event {
+            case .finished:
+                self?.quickLookController = nil
+            }
+        }
+        let controller = QuickLookController(viewModel: viewModel)
+        quickLookController = controller
+        navigationPresenter.present(controller.viewController)
+    }
+
+    private func presentWebViewController(with url: URL) {
+        let configuration = SFSafariViewController.Configuration()
+        configuration.entersReaderIfAvailable = true
+        let safariViewController = SFSafariViewController(url: url, configuration: configuration)
+        safariViewController.view.accessibilityIdentifier = "safari_root_view"
+        navigationPresenter.present(safariViewController)
+    }
+}
+
+// MARK: Chat model
+extension ChatCoordinator {
+    // swiftlint:disable function_body_length
```

Comment:
Perhaps since we are working on this file now, it would be nice to remove this and act upon the body length.

Reply:
Sure. 898d914a16fa8f40e4ab501a34cd89da741a5303